pkg/process/runner/runner.go

```diff
@@ -93,37 +96,46 @@ func (l *CheckRunner) RunRealTime() bool {
 }
 
 // NewRunner creates a new CheckRunner
-func NewRunner(config ddconfig.Reader, sysCfg *sysconfigtypes.Config, hostInfo *checks.HostInfo, enabledChecks []checks.Check, rtNotifierChan <-chan types.RTResponse) (*CheckRunner, error) {
+func NewRunner(
+	config ddconfig.Reader,
+	sysCfg *sysconfigtypes.Config,
+	hostInfo *checks.HostInfo,
+	enabledChecks []checks.Check,
+	rtNotifierChan <-chan types.RTResponse,
+) (*CheckRunner, error) {
 	runRealTime := !config.GetBool("process_config.disable_realtime_checks")
 
-	cfg := &checks.SysProbeConfig{}
+	sysProbeCfg := &checks.SysProbeConfig{}
 	if sysCfg != nil && sysCfg.Enabled {
 		// If the sysprobe module is enabled, the process check can call out to the sysprobe for privileged stats
 		_, processModuleEnabled := sysCfg.EnabledModules[sysconfig.ProcessModule]
-		cfg.ProcessModuleEnabled = processModuleEnabled
-		cfg.MaxConnsPerMessage = sysCfg.MaxConnsPerMessage
-		cfg.SystemProbeAddress = sysCfg.SocketAddress
-		cfg.GRPCServerEnabled = sysCfg.GRPCServerEnabled
+		sysProbeCfg.ProcessModuleEnabled = processModuleEnabled
+		sysProbeCfg.MaxConnsPerMessage = sysCfg.MaxConnsPerMessage
+		sysProbeCfg.SystemProbeAddress = sysCfg.SocketAddress
+		sysProbeCfg.GRPCServerEnabled = sysCfg.GRPCServerEnabled
 	}
 
-	for _, c := range enabledChecks {
-		if err := c.Init(cfg, hostInfo, false); err != nil {
-			return nil, err
-		}
-	}
```

Comment:
This was moved below to the `Run()` method to avoid initializing any of the checks unless the process-agent component is active. This is specifically to avoid calling the process check's Init() as it creates the workloadmeta extractor server.

The runner being a dependency for the agent component will have it's constructor invoked, but the agent component's OnStart will not call `Run()` if it's not setup to run.

Reply:
What is the `workloadmeta extractor server`? 
Note: I did some changes to workloadmeta [here](https://github.com/DataDog/datadog-agent/pull/23388) and more will come soon.

Reply:
The `workloadmeta extractor` is used for language detection of a process in the process-agent. The `workloadmeta extractor server` is a grpc server which provides the replicates the data from the process-agent to the core agent to provide language detection to the core agent.